routers/web/repo/setting.go

```diff
@@ -911,6 +911,18 @@ func CollaborationPost(ctx *context.Context) {
 		return
 	}
 
+	// find the owner team of the organization the repo belongs too and
+	// check if the user we're trying to add is an owner.
+	if ctx.Repo.Repository.Owner.IsOrganization() {
+		if isOwner, err := organization.IsOrganizationOwner(ctx, ctx.Repo.Repository.Owner.ID, u.ID); err != nil {
+			ctx.ServerError("IsOrganizationOwner", err)
+			return
+		} else if isOwner {
+			ctx.Redirect(setting.AppSubURL + ctx.Req.URL.EscapedPath())
```

Comment:
Might want to use a `ctx.Flash.Error` to notify why the collaborator wasn't added.

Reply:
flash added